docs/en/Image-Manipulation.md

```diff
@@ -0,0 +1,290 @@
+    
+    Task<ImageCompressResult<byte[]>> CompressAsync(
+        byte[] bytes,
+        [CanBeNull] string mimeType = null,
+        CancellationToken cancellationToken = default);
+}
+```
+
+Usage example:
+
+```csharp
+var result = await _imageCompressor.CompressAsync(
+    stream,
+    mimeType: "image/jpeg"
+);
+```
+
+## ImageCompressResult
+
+The `ImageCompressResult` is a generic class that is used to return the result of the image compression operations. It has the following properties:
+
+* `Result`: The compressed image stream or byte array.
+* `State`: The state of the compression operation. (See the [ProcessState](#processstate) section for more information.)
+
+## IImageCompressorContributor
+
+The `IImageCompressorContributor` is an interface that is used to implement a custom image compressor. It has the following method:
+
+```csharp
+public interface IImageCompressorContributor
+{
+    Task<ImageCompressResult<Stream>> TryCompressAsync(Stream stream, string mimeType = null, CancellationToken cancellationToken = default);
+    Task<ImageCompressResult<byte[]>> TryCompressAsync(byte[] bytes, string mimeType = null, CancellationToken cancellationToken = default);
+}
+```
+
+## ProcessState
+
+The `ProcessState` is an enum that is used to define the state of the image resize/compression operations. It has the following values:
+
+```csharp
+public enum ProcessState : byte
+{
+    Done = 1,
+    Canceled = 2,
+    Unsupported = 3,
+}
+```
+
+## Configuration
+
+### ImageResizeOptions
+
+`ImageResizeOptions` is used to configure the image resize system. It has the following properties:
+
+* `DefaultResizeMode`: The default resize mode. (Default: `ImageResizeMode.None`)
+
+## [Magick.NET](https://github.com/dlemstra/Magick.NET)
+
+Add the `Volo.Abp.Imaging.MagickNet` NuGet package to your project and use the `AbpImagingMagickNetModule` module.
+
+```csharp
+[DependsOn(typeof(AbpImagingMagickNetModule))]
+public class MyModule : AbpModule
+{
+    //...
+}
+```
+
+### Configuration
+
+#### MagickNetCompressOptions
+
+`MagickNetCompressOptions` is used to configure the Magick.NET image compression system. It has the following properties:
+
+* `OptimalCompression`: Indicates whether the optimal compression is enabled or not. (Default: `false`)
+* `IgnoreUnsupportedFormats`: Indicates whether the unsupported formats are ignored or not. (Default: `false`)
+* `Lossless`: Indicates whether the lossless compression is enabled or not. (Default: `false`)
+
+## [ImageSharp](https://github.com/SixLabors/ImageSharp)
+
+Add the `Volo.Abp.Imaging.ImageSharp` NuGet package to your project and use the `AbpImagingImageSharpModule` module.
+
+```csharp
+[DependsOn(typeof(AbpImagingImageSharpModule))]
+public class MyModule : AbpModule
+{
+    //...
+}
+```
+
+### Configuration
+
+#### ImageSharpCompressOptions
+
+`ImageSharpCompressOptions` is used to configure the ImageSharp image compression system. It has the following properties:
```

Comment:
It would be good to show a simple code example to configure one of these option values. Not everyone maybe familiar with the options pattern usage. Also, whenever you mention about options, add a link to the Options document, so people can understand what is the options pattern and how it works in general.